pyrenew_hew/ww_site_level_dynamics_model.py

```diff
@@ -0,0 +1,538 @@
+        numpyro.deterministic("r_subpop_t", r_subpop_t)
+
+        state_inf_per_capita = jnp.sum(self.pop_fraction * new_i_subpop, axis=1)
+        numpyro.deterministic("state_inf_per_capita", state_inf_per_capita)
+
+        # Hospital admission component
+        p_hosp_mean = self.p_hosp_mean_rv()
+        p_hosp_w_sd = self.p_hosp_w_sd_rv()
+        autoreg_p_hosp = self.autoreg_p_hosp_rv()
+
+        p_hosp_ar_init_rv = DistributionalVariable(
+            "p_hosp_ar_init",
+            dist.Normal(
+                0,
+                p_hosp_w_sd / jnp.sqrt(1 - jnp.pow(autoreg_p_hosp, 2)),
+            ),
+        )
+
+        p_hosp_ar_init = p_hosp_ar_init_rv()
+        p_hosp_ar_proc = ARProcess()
+        p_hosp_ar = p_hosp_ar_proc.sample(
+            noise_name="p_hosp_noise",
+            n=n_weeks_post_init,
+            autoreg=autoreg_p_hosp,
+            init_vals=p_hosp_ar_init,
+            noise_sd=p_hosp_w_sd,
+        )
+
+        ihr = jnp.repeat(
+            transformation.SigmoidTransform()(p_hosp_ar + p_hosp_mean),
+            repeats=7,
+        )[:n_datapoints]
+        numpyro.deterministic("ihr", ihr)
+
+        hosp_wday_effect_raw = self.hosp_wday_effect_rv()
+        inf_to_hosp = self.inf_to_hosp_rv()
+
+        hosp_wday_effect = tile_until_n(hosp_wday_effect_raw, n_datapoints)
+
+        potential_latent_hospital_admissions = compute_delay_ascertained_incidence(
+            p_observed_given_incident=1,
+            latent_incidence=state_inf_per_capita,
+            delay_incidence_to_observation_pmf=inf_to_hosp,
+        )[-n_datapoints:]
+
+        latent_hospital_admissions = (
+            potential_latent_hospital_admissions
+            * ihr
+            * hosp_wday_effect
+            * self.state_pop
+        )
+        numpyro.deterministic("latent_hospital_admissions", latent_hospital_admissions)
+
+        hospital_admission_obs_rv = NegativeBinomialObservation(
+            "observed_hospital_admissions", concentration_rv=self.phi_rv
+        )
+
+        if not is_predictive:
+            mu_obs_hosp = latent_hospital_admissions[self.hosp_times]
+        else:
+            mu_obs_hosp = latent_hospital_admissions
+
+        observed_hospital_admissions = hospital_admission_obs_rv(
+            mu=mu_obs_hosp,
+            obs=data_observed_hospital_admissions,
+        )
+
+        # wastewater component
+        if self.include_ww:
+            t_peak = self.t_peak_rv()
+            dur_shed = self.dur_shed_after_peak_rv()
+            s = get_vl_trajectory(t_peak, dur_shed, self.max_shed_interval)
+
+            def batch_colvolve_fn(m):
+                return jnp.convolve(m, s, mode="valid")
+
+            model_net_i = jax.vmap(batch_colvolve_fn, in_axes=1, out_axes=1)(
+                new_i_subpop
+            )[-n_datapoints:, :]
+            numpyro.deterministic("model_net_i", model_net_i)
+
+            log10_g = self.log10_g_rv()
+            model_log_v_ot = (
```

Comment:
This strikes me as a particularly confusing variable name. What are `v` and `ot` and why do we prefix with `model`?